### DISCUSSION TITLE
The callback parameter in the emit method does not work

### DISCUSSION THREAD

**Discussion by hitplum** [miguelgrinberg/python-socketio#1252] [Q&A]
My code is as follows：

```
def callback(*args, **kwargs):
    print('callback', args, kwargs)

@sio.event
async def connect(sid, environ):
    await sio.emit("my_message", data={"code": 1}, callback=callback, to=sid)
```

The request log as follows:
```
ZSUXVut3RRkWYrT0AAAA: Sending packet OPEN data {'sid': 'ZSUXVut3RRkWYrT0AAAA', 'upgrades': [], 'pingTimeout': 20000, 'pingInterval': 25000}
ZSUXVut3RRkWYrT0AAAA: Received request to upgrade to websocket
ZSUXVut3RRkWYrT0AAAA: Upgrade to websocket successful
ZSUXVut3RRkWYrT0AAAA: Received packet MESSAGE data 0
ZSUXVut3RRkWYrT0AAAA: Sending packet MESSAGE data 21["my_message",{"code":1}]
ZSUXVut3RRkWYrT0AAAA: Sending packet MESSAGE data 0{"sid":"cBLM9kuBTZjeB50WAAAB"}
service task canceled
ZSUXVut3RRkWYrT0AAAA: Sending packet PING data None
ZSUXVut3RRkWYrT0AAAA: Received packet PONG data
```
This callback method will never be executed, can you help me solve it.

- python-socketio==5.9.0
- python-engineio==4.7.1

**Comment by miguelgrinberg** [comment]
Please do not write issues on this project when you are just asking a question regarding your own application. The issues board is to track bugs in this package. The discussions board is for questions and seeking advice. Thank you. This time I have moved your question for you.

The callback is called after the other side acknowledges that it received the event. You have to look at the documentation of your client to find out how are acknowledgements sent.
